IdentityCore/src/cache/MSIDMacKeychainTokenCache.m

```diff
@@ -0,0 +1,660 @@
+  access and update credentials without further keychain prompts.
+
+Reference(s):
+* Apple Keychain Services: https://developer.apple.com/documentation/security/keychain_services?language=objc
+* Schema:
+https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview?path=%2FUnifiedSchema%2FSchema.md&version=GBdev
+
+*/
+
+static NSString *s_defaultKeychainGroup = @"com.microsoft.identity.universalstorage";
+static NSString *s_defaultKeychainLabel = @"Microsoft Identity Universal Storage";
+static MSIDMacKeychainTokenCache *s_defaultCache = nil;
+
+@interface MSIDMacKeychainTokenCache ()
+
+@property (readwrite, nonnull) NSString *keychainGroup;
+@property (readwrite, nonnull) NSDictionary *defaultAccountQuery;
+
+@end
+
+@implementation MSIDMacKeychainTokenCache
+
+#pragma mark - Public
+
++ (NSString *)defaultKeychainGroup
+{
+    return s_defaultKeychainGroup;
+}
+
+// Set the default keychain group
+//
+// Errors:
+// * @throw - attempt to change the default keychain group after being initialized
+//
++ (void)setDefaultKeychainGroup:(NSString *)defaultKeychainGroup
+{
+    MSID_TRACE;
+
+    if (s_defaultCache)
+    {
+        MSID_LOG_ERROR(nil, @"Failed to set default keychain group, default keychain cache has already been instantiated.");
+
+        @throw @"Attempting to change the keychain group once AuthenticationContexts have been created or the default keychain cache has been retrieved is invalid. The default keychain group should only be set once for the lifetime of an application.";
+    }
+
+    MSID_LOG_INFO(nil, @"Setting default keychain group.");
+    MSID_LOG_INFO_PII(nil, @"Setting default keychain group to %@", defaultKeychainGroup);
+
+    if ([defaultKeychainGroup isEqualToString:s_defaultKeychainGroup])
+    {
+        return;
+    }
+
+    if (!defaultKeychainGroup)
+    {
+        defaultKeychainGroup = [[NSBundle mainBundle] bundleIdentifier];
+    }
+
+    s_defaultKeychainGroup = [defaultKeychainGroup copy];
+}
+
++ (MSIDMacKeychainTokenCache *)defaultKeychainCache
+{
+    static dispatch_once_t s_once;
+
+    dispatch_once(&s_once, ^{
+        s_defaultCache = [MSIDMacKeychainTokenCache new];
+    });
+
+    return s_defaultCache;
+}
+
+#pragma mark - init
+
+// Initialize with defaultKeychainGroup
+- (nonnull instancetype)init
+{
+    return [self initWithGroup:s_defaultKeychainGroup];
+}
+
+// Initialize with a keychain group
+//
+// @param keychainGroup Optional. If the application needs to share the cached tokens
+// with other applications from the same vendor, the app will need to specify the
+// shared group here. If set to 'nil' the main bundle's identifier will be used instead.
+//
+- (nullable instancetype)initWithGroup:(nullable NSString *)keychainGroup
+{
+    MSID_TRACE;
+
+    self = [super init];
+    if (self)
+    {
+        if (!keychainGroup)
+        {
+            keychainGroup = [[NSBundle mainBundle] bundleIdentifier];
+        }
+
+        if (MSIDKeychainUtil.teamId)
```

Comment:
Note that some updates will be necessary for MSIDKeychainUtil to work on macOS. Might make sense to have a separate mac keychain util for teamId retrieval or subclass.

Reply:
split this out into a separate follow-up task